lib/jsonapi/serializable/resource/caching.rb

```diff
@@ -0,0 +1,48 @@
+module JSONAPI
+  module Serializable
+    class Resource
+      # Extension for caching dynamic attributes.
+      #
+      # @usage
+      #   class SerializableUser < JSONAPI::Serializable::Resource
+      #     extend JSONAPI::Serializable::Resource::Caching
+      #
+      #     # Optionally specify a subset of attributes to be cached.
+      #     cached_attributes :name, :email
+      #     ...
+      #   end
+      module Caching
```

Comment:
this seems to only be per-render caching.
maybe a subsequent PR to allow cross-render cache?